src/util/Util.ts

```diff
@@ -15,45 +15,112 @@
  */
 /** @ignore *//** */
 
-import * as assert from 'assert';
+import * as _assert from 'assert';
 import * as Long from 'long';
 import * as Path from 'path';
 import {BigDecimal, IllegalStateError, LocalDate, LocalDateTime, LocalTime, MemberImpl, OffsetDateTime, UUID} from '../core';
 import {MemberVersion} from '../core/MemberVersion';
 import {BuildInfo} from '../BuildInfo';
+import {AssertionError} from 'assert';
+
+// Used to assert only in development mode
+const IS_DEVELOPMENT_MODE = process.env.HZ_NODEJS_ENV === 'development';
+
+/** @internal */
+export const assert : (value: any, message?: string | Error) => void = _assert;
+
+/** @internal */
+export const assertCond : (value: any, message?: string | Error) => void = IS_DEVELOPMENT_MODE ? _assert : () => {};
+
+/** @internal */
+export const assertNotNullCond : (v: any) => void = IS_DEVELOPMENT_MODE ? assertNotNull : () => {};
 
 /** @internal */
 export function assertNotNull(v: any): void {
-    assert.notStrictEqual(v, null, 'Non null value expected.');
+    if (v === null) {
+        throw new AssertionError({
+            message: 'Non null value expected.',
+            expected: null,
+            actual: v
+        });
+    }
 }
 
 /** @internal */
 export function assertArray(x: any): void {
-    assert(Array.isArray(x), 'Should be array.');
+    const isArray = Array.isArray(x);
```

Comment:
TBH, the old version looks more compact. I can say a similar thing for other assertions as well. What is the reason we changed them?